### ISSUE TITLE
Feature Request: Ability to set own Sleep Timer and Option "Sleep at the end of the Track"

### ISSUE THREAD

**Issue by TheLux83** [kaaholst/android-squeezer#758]
Hey there.

After a long search for a good media server for listening to music and ambient sounds, I've found LMS and Squeezer to be a good combination :-)

But there is one (I hope little) thing, that hold me back from using it completely. Squeezer has a sleep timer function, which is something not many apps have, but it's limited to 90 minutes.
We're using ambient sounds for sleeping and my SO unfortunately needs a long time until she falls asleep. At the moment we're using spotify which has the option to sleep at the end of the track. Since we're using 8 hours ambient sounds, that is an okay'ish solution.
What I wish, is the ability to set my own sleep time, or if this isn't possible, maybe add a few more options like 2 hours and 3 hours.
I also would love to see a "sleep at the end of the track" if possible.

TL;DR: Is it possible to add a manual timer or higher sleep timer like 2 and 3 hours and an option to "sleep at the end of the track"?

Thanks for all the good work here. The app is amazing! This would be the last thing missing and I can switch my "nightly driver" in our bedroom :-)

**Comment by kaaholst** [comment]
Hi and thanks for the feedback.

Currently you can set the sleep timer in 3 ways.

1. From the 3-dot menu to the right of each player on the players screen.
2. From the 3-dot on the top right corner of the screen for the selected player.
3. Via Home / Settings / Sleep for the active player.

For each of these ways there will be an option to sleep at end of song, if the player is currently playing.

Option one and two are implemented in the Squeezer app, so we can easily add options for 2 and 3 hours (120 and 180 minutes). The command for the server just takes a number of minutes, so we could even add the possible to enter the timer duration, albeit not as easy.

Regarding options from the Home menu, the deal is that the Squeezer app, queries LMS for what can be done for the selected player, and displays these options (they call them menus). While it is technically possible to modify  the returned items, this would require more work and more maintenance of the Squeezer app. The benefit of the current solution, is that new options or plugins in LMS will automatically show up in Squeezer, without any modification of the app.

**Comment by TheLux83** [comment]
I didn't know, that there are 3 ways to set the sleep timer.

I'm totally okay with the sleep at the end of song thing, if this is the easiest to implement for all three ways. 
I don't want you to change the way, the app interacts with LMS, so if this will be the only option, I'll be okay with it, too. But for me I'm also happy, if I only have the 2 and 3 hours option for the first two ways. 

Since I'm not familiar with coding Android Apps and therefore not knowing how hard it is to implement these changes, I'm happy with everything I can get ^^

**Comment by kaaholst** [comment]
Rereading this I see there may be some confusion, so here is a recap and a suggestion.

Sleep at end of track is already present in Squeezer, the option is only visible if the player is currently playing though.

For the next version I will implement a "manual sleep duration" option, which when selected will prompt for a number of minutes until the player will sleep. It will default to sleep duration the manual option was previously selected.

How does that sound?

**Comment by TheLux83** [comment]
That sounds pretty awesome :-)
I understood, that there is already a "Sleep at end of track" option and I'm using it every night. At the moment I forward my ambient sounds until there is only that amount of time left, I want it to play and it works pretty well.

But having the option to select my own timer with a prompt would be pretty awesome. Especially if one has Ambient sounds, that don't go long enough, or if one has an audiobook with chapters that are only a few minutes long.

**Comment by Snowman-25** [comment]
Fixed as of 2.30, AFAIK

**Comment by kaaholst** [comment]
> Fixed as of 2.30, AFAIK

You're right, this is available from 2.3.0, you beat me to it!

Closing this, and thanks for the feedback.
